inc/generation.class.php

```diff
@@ -105,17 +105,12 @@ static function preItemAdd(CommonDBTM $item) {
     * @override CommonDBTM::preItemUpdate
     */
    static function preItemUpdate(CommonDBTM $item) {
-      if (!Session::haveRight("plugin_geninventorynumber", UPDATE)) {
-         return;
-      }
-
-      if (PluginGeninventorynumberConfig::isGenerationActive()
-         && PluginGeninventorynumberConfigField::isActiveForItemType(get_class($item))
-            && !isset($item->input['massiveaction'])) {
-
-         if (isset($item->fields['otherserial'])
-            && isset($item->input['otherserial'])
-               && $item->fields['otherserial'] != $item->input['otherserial']) {
+      $active = PluginGeninventorynumberConfig::isGenerationActive() &&
+          PluginGeninventorynumberConfigField::isActiveForItemType(get_class($item));
+      if ($active && !isset($item->input['massiveaction'])) {
```

Comment:
`massiveaction` is the generic name of submit button is massive actions. I may be wrong, but I guess checking this flag will not prevent ability to use the generic `update` massive action to change the serial number.
I think a specific flag should be added in the plugin process (in `doMassiveUpdate` I guess) to be sure to check on something defined by the plugin itself. And to ensure noone will try to hack this by sending this in a HTTP query payload, this flag should be set on the item itself (e.g. `$tmp->_plugin_geninventorynumber_allowed = true; $tmp->update($values)`